src/v2/providers/storage.ts

```diff
@@ -382,7 +389,7 @@ export function getOptsAndBucket(
     bucket = bucketOrOpts;
     opts = {};
   } else {
-    bucket = bucketOrOpts.bucket || firebaseConfig().storageBucket;
+    bucket = bucketOrOpts.bucket || firebaseConfig()?.storageBucket;
```

Comment:
This was the NPE culprit. firebaseConfig() is nullable, so attempting to access `storageBucket` throw an exception.